Add appCustomVersion Checks

Skinflint opened the pull request:
Add checks for the appCustomVersion function in the Installomator labels to check if an app is installed, with a fallback to the pkgutil check if an app location was not found in the appCustomVersion.

Add a check for .plugin since plugins like JRE8 do not require a signature unlike a .app. This checks the files inside the {PluginPath}Contents/Home/Bin folder for their TeamID.

Fix the installmode bug that caused the script to always run in install mode.

Quiet some lines if the script is run in quiet mode.

Skinflint committed:
Incorporate appCustomVersion checks

Add checks for the appCustomVersion function in the Installomator labels to check if an app is installed. Also add a check for plugins to bypass appVerify since plugins do not require a signature unlike apps.

Skinflint committed:
Plugin check

Update the script to check for signatures of a plugin instead of bypass all checks on a plugin. Also fix the installmode check and make a few more things quiet when the quiet flag is used.

Skinflint committed:
Logic Updates

Update some logic to check if a command line arguments were accidentally passed twice instead of just once. Also update the displayconfig at the end to still show when running in quiet mode unless the write flag was also set.

Skinflint committed:
Log rolling

Added Log rolling if file is greater than 1 MB by default. Also set the maximum number of rolled files to 5 by default. These 5 files start at Patchomator.log.1 and go to Patchomator.log.5 for the oldest logs before getting purged.
Cleaned up some logging and exiting code to make things consistent.

Skinflint commented:
Added some log rolling and some logic updates as well to the script.

Skinflint committed:
Fix Case issue

Fixed the caffexit call to the finishAndExit function.

Skinflint committed:
Install and Write Check

Added a check for if the --install and --write arguments are passed together and if so, default to the --install without overwriting the config file with a blank file.

Skinflint committed:
Remove extra finished output

Removed the extra finished output on an install, as well as changed the errors output to infoOut in order to honor quiet mode.

Mac-Nerd commented:
These are great. Working on reviewing/merging.

Mac-Nerd commented:
For .plugin (and other future non-app updates) does `--everywhere` catch those, or not yet?

Mac-Nerd reviewed (changes requested):
Can't remove this "finished" line - it's from a user request for something they can search on to check if the script is still running, or stuck.

Mac-Nerd commented on patchomator.sh:
```diff
@@ -667,15 +681,15 @@ SCRIPT_EOF
 		infoOut "${appPath} already linked to label ${exists}."
 		if [[ ${#noninteractive} -eq 1 ]]
 		then
-			echo "\t${BOLD}Skipping.${RESET}"
```
this one is fine.

Mac-Nerd commented on patchomator.sh:
```diff
@@ -667,15 +681,15 @@ SCRIPT_EOF
+			infoOut "\t${BOLD}Skipping.${RESET}"
 			return
 		else
-			echo -n "${BOLD}Replace label ${exists} with $foundLabel? ${YELLOW}[y/N]${RESET} "
```
if the [y/N] questions don't echo in quiet mode, then the script will hang waiting for an answer on the `read`, right? Also, I want to avoid things like FirefoxIntl automatically replacing Firefox.

Mac-Nerd reviewed (approved):
Thanks. This is much cleaner for command-line option parsing.

Skinflint commented:
I believe that all of the finished statements still happen, they are just now all put in a function so that they can maintain consistency if you ever want to change the verbage or add something else after everything is finished, but before the script completes. 

I do see the problem with the question on the do you want to replace the label, so I will change that one back to an echo instead of going to infoOut.

The --everywhere still won't look at anything put .app files since the search is searching for $appName. I didn't want to change that function without more time to delve into if that would work with other file extensions. 

I will make the changes and then commit again.

Mac-Nerd reviewed (commented):
Note to self: Should check to see if the system log-rolling can be configured to include Patchomator.log so it can comply with whatever MDM or user-defined preferences.

Skinflint committed:
Fix Replace Question

Fixed the echo for the question of whether or not to update the found label. Changed some if statements to use an interger check instead of -eq 1 in case the command line argument was added twice.

Mac-Nerd commented:
FYI - if you're on Macadmins Slack, I'm @option8. DM me.

Skinflint committed:
Fix variable in interactive mode

Fixed the variable in interactive mode that added the last label in the list to the array instead of the current label that is being processed.
